cpp/src/utilities/device_operators.cuh

```diff
@@ -76,15 +76,35 @@ struct DeviceCount {
 
 /* @brief binary `min` operator */
 struct DeviceMin {
-  template <typename T>
+  template <typename T,
+            typename std::enable_if_t<!std::is_same<T, cudf::string_view>::value>* = nullptr>
   CUDA_HOST_DEVICE_CALLABLE T operator()(const T& lhs, const T& rhs) {
     return std::min(lhs, rhs);
   }
 
-  template <typename T>
+  template <typename T,
+            typename std::enable_if_t<!std::is_same<T, cudf::string_view>::value>* = nullptr>
   static constexpr T identity() {
     return std::numeric_limits<T>::max();
   }
+
+  template <typename T,
+            typename std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
+  CUDA_HOST_DEVICE_CALLABLE T operator()(const T& lhs, const T& rhs) {
+    if (lhs.empty())
+      return rhs;
+    else if (rhs.empty())
+      return lhs;
+    else
+      return std::min(lhs, rhs);
+    //return lhs <= rhs ? lhs : rhs;
+  }
+
+  template <typename T,
+            typename std::enable_if_t<std::is_same<T, cudf::string_view>::value>* = nullptr>
+  static constexpr T identity() {
+    return T{nullptr, 0};
```

Comment:
Why is the identity a null string view?

Reply:
it is designed for `min` and `max` of string_view. I moved this identity inside `min` and `max` operator as template specialization. 
For `min`, identity is largest possible value. For string, it is not efficient to create an arbitrary large string, so, I added special case for handling null string view.
For `max`, identity is smallest possible value which is an empty string.